hexrd/fitting/utils.py

```diff
@@ -17,7 +19,8 @@ def _parameter_arg_constructor(pdict, pargs):
 
 
 def _extract_parameters_by_name(params, pname_root):
-    return [s for s in params.keys() if s.__contains__(pname_root)]
+    # return [s for s in params.keys() if s.__contains__(pname_root)]
+    return fnmatch.filter(list(params.keys()), pname_pattern)
```

Comment:
@joelvbernier If you wanted to continue supporting substring matches, one option would be to just add `*` to the ends of the pattern, like so:

`f'*{pname_pattern}*'`

Then substrings would still match, and you can do `*` inside the string as well.

Alternatively, you could add an argument such as `match_substring=True` where, if it is true, then you add the `*` to the ends of the pattern.

Reply:
of course -- bracketing with '*' might be the most elegant solution :-)